### PR TITLE
Put failed events on an SQS DLQ

### PR DESCRIPTION
**Pull request by postsa**
### What does this pull request do?

* Created an SQS queue to be used as a DLQ
* Unhandled exceptions will go onto the DLQ for alerting purposes.

### Ticket
[Salesforce Application State Changed Event Consumer: Send failed events to a custom SQS DLQ](https://app.clubhouse.io/guild/story/51736/salesforce-application-state-changed-event-consumer-send-failed-events-to-a-custom-sqs-dlq)

#### Required before merge:
- [X] Has tests
- [X] Passes CircleCI
- [X] Makes its author(s) proud

### CODE REVIEW

File: src/sqs_dlq.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+class SqsDlq {
+    constructor(config) {
+        this.config = config;
+        this.sqs = new SQS({})
+    }
+
+    buildMessage(error, event) {
+        return {
+            MessageAttributes: {
+                "Event": {
```
Comment (travishaby) [inline src/sqs_dlq.js]:
```suggestion
                Event: {
```

File: src/sqs_dlq.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                    DataType: "String",
+                    StringValue: JSON.stringify(event)
+                },
+                "Stacktrace": {
```
Comment (travishaby) [inline src/sqs_dlq.js]:
```suggestion
                Stacktrace: {
```

File: src/sqs_dlq.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        }
+    }
+
+    async put(error, event) {
+        const message = this.buildMessage(error, event);
+        try {
+            console.log(`An error occurred while processing an event. ${error}.`);
+            await this.sqs.sendMessage(message);
+        } catch(ex) {
+            console.log(`Error putting failed event on DLQ ${ex}.`)
```
Comment (travishaby) [inline src/sqs_dlq.js:32]: to catch failed publishes to the DLQ, should we alert on this string or is there some kind of AWS event like "sqs.sendMessage.failure" we can watch for ?

File: template.yaml [hunk 4]
```diff
@@ -61,7 +74,9 @@ Resources:
               - 'kinesis:ListShards'
               - 'kinesis:ListStreams'
               - 'kinesis:SubscribeToShard'
-            Resource: '*'
+            Resource:
```
Comment (travishaby) [inline template.yaml:77]: 🤙

**Review by travishaby — APPROVED** [review]
Looks good! Just some minors comments

File: src/sqs_dlq.js (hunk 3 above)
Comment (postsa) [reply src/sqs_dlq.js:32]: Yea we'd have to alert on the log. I don't think the SQS itself can be aware of all failures to send messages to it.

**Pull request merged**